### PR TITLE
exchange manager documentation and Add() comment

### PR DESCRIPTION
**Pull request by romanornr**
# PR Description

exchange manager documentation and Add() comment
Some extra comments might clear up and help new users trying to understand 

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: engine/exchange_manager.go [hunk 1]
```diff
@@ -66,7 +66,11 @@ func SetupExchangeManager() *ExchangeManager {
 	}
 }
 
-// Add adds or replaces an exchange
+// Add function of the ExchangeManager is used to add or replace an exchange.
+// If the exchange is a nil value, it will return and not perform the access to the Lock and Unlock functions.
+// If the exchange is not nil, the Lock function will be called, and when the Unlock function is called, the exchange has been accessed by a remote interface and is no longer nil.
```
[the remark this answers is not in the document]
Comment (shazbert) [reply engine/exchange_manager.go:71]: From the logs It looks like you might have made a custom exchange manager instance without instantiating the map.  SetupExchangeManager() func is usually called first, but seeing as you are doing custom code, it's hard to make a determination and help you out.

[the remark this answers is not in the document]
Comment (romanornr) [reply engine/exchange_manager.go:71]: I found out a couple hours what I did wrong or missed! 

I had to do it like this within a struct ExchangeManager: *engine.SetupExchangeManager(),

[the remark this answers is not in the document]
Comment (romanornr) [reply engine/exchange_manager.go:71]: Added another commit
https://github.com/thrasher-corp/gocryptotrader/pull/767/commits/c0a1a4a0cdb23529c9d8350695f27edde9d0dc39

might review this as well 

Have another commit incoming but might get a little bit messy with 3 commits 
Have to check out how I can add it as a separate pull request

File: engine/engine.go [hunk 2]
```diff
@@ -735,7 +739,10 @@ func (bot *Engine) LoadExchange(name string, wg *sync.WaitGroup) error {
 			var pairs currency.Pairs
 			pairs, err = exchCfg.CurrencyPairs.GetPairs(assets[x], false)
 			if err != nil {
-				return err
+				gctlog.Errorf(gctlog.ExchangeSys,
```
Comment (shazbert) [inline engine/engine.go:742]: Just curious, what happened here that required this change?

File: engine/engine.go [hunk 3]
```diff
@@ -706,8 +706,12 @@ func (bot *Engine) GetExchanges() []exchange.IBotExchange {
 	return bot.ExchangeManager.GetExchanges()
 }
 
-// LoadExchange loads an exchange by name. Optional wait group can be added for
-// external synchronization.
+// LoadExchange creates an exchange object for the loaded exchange by calling ExchangeManager.NewExchangeByName.
```
Comment (shazbert) [inline engine/engine.go:709]: I prefer keeping comments concise, and the actual code be annotated instead.

**Review by shazbert — CHANGES_REQUESTED** [review]
Appreciate the effort you have put into this PR, I do have some questions though. The comments are very verbose (1) and they exceed common line lengths (2). This is just my view though, am happy for more discussion and I understand there are some people that would appreciate this change.

**Comment by thrasher-** [comment]
Closed due to inactivity

**Pull request closed without merging**